### PR TITLE
Enhancement/3531 user feedback survey 300ms

### PR DESCRIPTION
**Pull request by danielgent**
## Summary

Addresses issue #3531

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/surveys/__fixtures__/index.js [hunk 1]
```diff
@@ -147,6 +147,64 @@ export const multiQuestionSurvey = {
 					],
 				},
 			},
+			{
+				question_ordinal: 3,
```
Comment (danielgent) [inline assets/js/components/surveys/__fixtures__/index.js:151]: makes the UX better to play around with in storybook

File: assets/js/components/surveys/CurrentSurvey.js [hunk 2]
```diff
@@ -195,6 +206,7 @@ export default function CurrentSurvey() {
 		<Slide direction="up" in={ animateSurvey } onExited={ handleAnimationOnExited }>
 			<div className="googlesitekit-survey">
 				<SurveyQuestionComponent
+					key={ currentQuestion.question_text }
```
Comment (danielgent) [inline assets/js/components/surveys/CurrentSurvey.js:209]: As per my comment here https://github.com/google/site-kit-wp/issues/3531#issuecomment-873076129

File: assets/js/components/surveys/CurrentSurvey.test.js [hunk 3]
```diff
@@ -218,7 +230,7 @@ describe( 'CurrentSurvey', () => {
 
 		fireEvent.click( getByLabelText( 'Delighted' ) );
 
-		expect(
+		await waitFor( () =>	expect(
```
Comment (eugene-manuilov) [inline assets/js/components/surveys/CurrentSurvey.test.js:233]: Please, don't use tabs within the line. We use tabs only for indentation.
```suggestion
		await waitFor( () => expect(
```

File: assets/js/components/surveys/CurrentSurvey.js (hunk 2 above)
Comment (eugene-manuilov) [reply assets/js/components/surveys/CurrentSurvey.js:209]: Nice catch! Thanks!

File: assets/js/components/surveys/CurrentSurvey.js [hunk 4]
```diff
@@ -70,23 +73,31 @@ export default function CurrentSurvey() {
 	const currentQuestion = questions?.find( ( { question_ordinal: questionOrdinal } ) => questionOrdinal === currentQuestionOrdinal );
 
 	const answerQuestion = useCallback( ( answer ) => {
-		sendSurveyEvent( 'question_answered', {
-			// eslint-disable-next-line camelcase
-			question_ordinal: currentQuestion?.question_ordinal,
-			answer,
-		} );
+		if ( ! hasAnsweredQuestion ) {
+			setHasAnsweredQuestion( true );
```
Comment (eugene-manuilov) [inline assets/js/components/surveys/CurrentSurvey.js:77]: Let's return early to reduce nesting:
```suggestion
		if ( hasAnsweredQuestion ) {
			return;
		}
		
		setHasAnsweredQuestion( true );
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @danielgent. In general looks good to me. Left two small comments and it should be good to go. Could you please take a look at it?

File: assets/js/components/surveys/CurrentSurvey.js (hunk 4 above)
Comment (danielgent) [reply assets/js/components/surveys/CurrentSurvey.js:77]: That's a nice change :smile:

**Review by eugene-manuilov — APPROVED** [review]
Awesome! Thanks, @danielgent!

**Pull request merged**